category_encoders/hashing.py

```diff
@@ -238,18 +161,64 @@ def _transform_single_cpu(self, X, override_return_df=False):
         if not list(self.cols):
             return X
 
-        X = self.hashing_trick(X, hashing_method=self.hash_method, N=self.n_components, cols=self.cols)
-
-        if self.drop_invariant:
-            X = X.drop(columns=self.invariant_cols)
+        X = self.hashing_trick(
+            X, 
+            hashing_method=self.hash_method,
+            N=self.n_components, 
+            cols=self.cols, 
+        )
+        
+        return X
 
-        if self.return_df or override_return_df:
-            return X
-        else:
-            return X.to_numpy()
+    @staticmethod
+    def hash_chunk(shm_result, np_df, N, shm_offset):
+        for i, row in enumerate(np_df):
+            for val in row:
+                if val is not None:
+                    hasher = hashlib.md5()
+                    hasher.update(bytes(str(val), 'utf-8'))
+                    column_index = int.from_bytes(hasher.digest(), byteorder='big') % N
```

Comment:
I will add a comment! The `digest` version is about 30% faster than the `hexdigest` version. On my machine:

```python
> import hashlib
> hasher = hashlib.md5()
> hasher.update("abdcde1234")
> %timeit int(hasher.hexdigest(), 16)
659 ns ± 29.5 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
> %timeit int.from_bytes(hasher.digest(), byteorder='big')
518 ns ± 11.5 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

The byte order is machine dependant only if the data comes directly from the machine. In that case, the `digest` should not be machine dependant. I tried it on
- my system (MacOS Monterey little endian) 
```python
> import sys
> import hashlib
> sys.byteorder
'little'
> hasher = hashlib.md5()
> hasher.update("abc")
> int.from_bytes(hasher.digest(), endian='big')
191415658344158766168031473277922803570
> int(hasher.hexdigest(), 16)
191415658344158766168031473277922803570
```
- a docker system (`s390x/ubuntu` image, big endian):
```python
> import sys
> import hashlib
> sys.byteorder
'big'
> hasher = hashlib.md5()
> hasher.update("abc")
> int.from_bytes(hasher.digest(), endian='big')
191415658344158766168031473277922803570
> int(hasher.hexdigest(), 16)
191415658344158766168031473277922803570
```

The result is the same.

Reply:
Ok that makes sense.

Which python/hashlib version are you using btw?
Running `hasher.update("abc")` results in a `TypeError` on my machine since it requires (as per documentation) bytes not strings

Reply:
I copied the wrong line from my shell! I updated my comment above.